pkg/syncer/metadatasyncer.go

```diff
@@ -966,6 +966,13 @@ func ReloadConfiguration(metadataSyncer *metadataSyncInformer, reconnectToVCFrom
 						}
 						vcenter.Config = newVCConfig
 					}
+					if !found {
+						// If the new VC host is not found, add its entry in
+						// volumeOperationsLock, cnsCreationMap and cnsDeletionMap.
+						cnsCreationMap[newVCConfig.Host] = make(map[string]bool)
```

Comment:
Can we have a lock between `full sync` and `reloadconfig` to protect `cnsCreationMap` and `cnsDeletionMap`.
only one should be operating on `cnsCreationMap` and `cnsDeletionMap` at a time.

If fullsync is running, lets hold `ReloadConfiguration` and if `ReloadConfiguration` is going on, lets hold fullsync